kite-data/kite-data-core/src/main/java/org/kitesdk/data/DatasetDescriptor.java

```diff
@@ -935,6 +936,21 @@ public DatasetDescriptor build() {
         columnMapping = copiedMapping;
       }
 
+      // Parse for properties in the entity schema and merge them with
+      // properties defined. If a property is defined using this builder and
+      // is also in the schema, the value set in the builder will take precedence
+      if(EntityPropertiesParser.hasEmbeddedProperties(schema)) {
```

Comment:
Hmm. When schema directory is migrated using the SchemaTool, either all the needed meta data is embedded inside the schema or I guess we would need a concept of a schema directory structure and define file names to read for these meta-data (paritions.json, schema.avro, entity.properties etc). The second approach is whole lot of work involved I guess. COTT team uses the migrateDirectory in SchemaTool to evolve schemas.